src/Binance/Signer.cpp

```diff
@@ -80,8 +80,13 @@ std::vector<uint8_t> Signer::encodeOrder() const {
 }
 
 std::vector<uint8_t> Signer::encodeSignature(const std::vector<uint8_t>& signature) const {
-    auto key = PrivateKey(input.private_key());
-    auto publicKey = key.getPublicKey(TWPublicKeyTypeSECP256k1);
+    PublicKey publicKey(Data(), TWPublicKeyTypeSECP256k1);
+    if (input.private_key().length() > 0) {
+        auto key = PrivateKey(input.private_key());
+        publicKey = key.getPublicKey(TWPublicKeyTypeSECP256k1);
+    } else {
+        publicKey = PublicKey(input.public_key(), TWPublicKeyTypeSECP256k1);
```

Comment:
you didn't check `public_key`

Reply:
I moved it out of protobuf and made it a parameter as @vikmeup suggested. The go code will verify its a valid secp256k1 pubkey